src/Adapter/InMemoryAdapter.php

```diff
@@ -163,18 +168,22 @@ public function loadEvents(
 
     public function loadEventsReverse(
         StreamName $streamName,
-        array $metadata = [],
         int $fromNumber = 0,
-        int $count = null
+        int $count = null,
+        MetadataMatcher $metadataMatcher = null
     ): Iterator {
         if (! isset($this->streams[$streamName->toString()])) {
             return new ArrayIterator();
         }
 
+        if (null === $metadataMatcher) {
+            $metadataMatcher = new MetadataMatcher();
+        }
+
         $streamEvents = [];
 
         foreach ($this->streams[$streamName->toString()]['events'] as $index => $streamEvent) {
```

Comment:
I don't know. Just submit a PR and make a suggestion, we'll see when we have it.